src/pages/ViewEmployee/ViewEmployee.tsx

```diff
@@ -1,49 +1,98 @@
 import { useParams } from "react-router-dom";
 import SelectedSkills from "../../components/SelectedSkills/SelectedSkills";
 import DivWrapper from "./styles";
-import { filterArray } from "../../utils/filterArray";
-import { useEmployeeContext } from "../../contexts/EmployeeContext";
+import { useState, useEffect } from "react";
+import { getData } from "../../core/api";
+import { IEmployee } from "../../components/Table/types";
+import { CircularProgress } from "@mui/material";
+import { toast } from "react-toastify";
 
 const ViewEmployee = () => {
   const { employeeId } = useParams();
-  const {employeesData} = useEmployeeContext();
-  const [employee] = filterArray(employeesData, { id: [Number(employeeId)] });
+  const [loading, setLoading] = useState(true);
+  const [employee, setEmployee] = useState<IEmployee>({ skills: [] });
+  const fetchCurrentEmployeeData = async () => {
+    setLoading(true);
+    try {
+      const response = (await getData(`employee/${employeeId}`)).data.data;
+      setEmployee({
+        ...response,
+        department: response.department
+          ? response.department.department
+          : "N/A",
+        role: response.role ? response.role.role : "N/A",
+      });
+    } catch (error) {
+      toast.error(
+        `Employee details could not be fetched from server.`,
+        {
```

Comment:
All toaster property will be consistant right? the why don't you make it as a different component with props that accept success,failure or warning?